lib/status/status.go

```diff
@@ -33,16 +34,20 @@ import (
 	"github.com/gravitational/gravity/lib/storage"
 	"github.com/gravitational/gravity/lib/utils"
 
+	httptransport "github.com/go-openapi/runtime/client"
 	"github.com/gravitational/roundtrip"
 	pb "github.com/gravitational/satellite/agent/proto/agentpb"
 	"github.com/gravitational/satellite/monitoring"
 	"github.com/gravitational/trace"
+	alertmanager "github.com/prometheus/alertmanager/api/v2/client"
+	"github.com/prometheus/alertmanager/api/v2/client/alert"
+	"github.com/prometheus/alertmanager/api/v2/models"
 	"github.com/sirupsen/logrus"
 )
 
 // FromCluster collects cluster status information.
 // The function returns the partial status if not all details can be collected
-func FromCluster(ctx context.Context, operator ops.Operator, cluster ops.Site, operationID string) (status *Status, err error) {
+func FromCluster(ctx context.Context, operator ops.Operator, cluster ops.Site, operationID, dnsAddr string) (status *Status, err error) {
```

Comment:
I'm not super happy about this, I wanted to pass the entire environment for grabbing the status, but it seems to create a circular dependency tree. So for now only passing the dnsAddress in use.

Reply:
Yeah, we actually pass DNS config around quite a bit in different place around the code which is unfortunate. I would at least convert these arguments to a struct (`StatusConfig` or something) so you don't have to pass empty string `""` where it's not needed.

```go
type StatusConfig struct {
  Operator ops.Operator
  Cluster ops.Site
  OperationID String
  DNSAddress string
}

func FromCluster(ctx context.Context, conf StatusConfig) ...
```